var/spack/repos/builtin/packages/groff/package.py

```diff
@@ -35,6 +38,12 @@ class Groff(AutotoolsPackage, GNUMirrorPackage):
     # patch('gropdf.patch')
     parallel = False
 
+    @classmethod
+    def determine_version(cls, exe):
+        output = Executable(exe)('--version', output=str, error=str)
+        match = re.search(r'groff.*version\s+(\S+)', output)
```

Comment:
The output for my system groff looks like:
```console
$ groff --version
GNU groff version 1.19.2
Copyright (C) 2004 Free Software Foundation, Inc.
GNU groff comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of groff and its subprograms
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.

called subprograms:

GNU grops (groff) version 1.19.2
GNU troff (groff) version 1.19.2
```
Since your regex matches multiple lines, I would maybe make it more specific, like:
```suggestion
        match = re.search(r'GNU groff version\s+(\S+)', output)
```